dojo/api_v2/serializers.py

```diff
@@ -1485,6 +1509,12 @@ def get_engagement(self, obj):
         )
 
     def validate(self, data):
+        def validate_findings_have_same_engagement(finding_objects: List[Finding]):
+            engagements = finding_objects.values_list('test__engagement__id', flat=True).distinct().count()
+            if engagements > 1:
+                msg = "You are not permitted to add findings to a distinct engagement"
```

Comment:
I think something like this better matches the intent here?

```suggestion
                msg = "You are not permitted to add findings from multiple engagements"
```